src/blop/dofs.py

```diff
@@ -80,57 +90,80 @@ class DOF:
+                self.search_domain = tuple(self.search_domain)
+                if len(self.search_domain) != 2:
+                    raise ValueError("'search_domain' must be a 2-tuple of floats.")
+                if self.search_domain[0] > self.search_domain[1]:
+                    raise ValueError("The lower search bound must be less than the upper search bound.")
+
+            if self.trust_domain is not None:
+                self.trust_domain = tuple(self.trust_domain)
+                if not self.read_only:
+                    if (self.search_domain[0] < self.trust_domain[0]) or (self.search_domain[1] > self.trust_domain[1]):
+                        raise ValueError("Trust domain must be larger than search domain.")
+
+            if self.log:
+                if not self.search_domain[0] > 0:
+                    raise ValueError("Search domain must be strictly positive if log=True.")
+
+            if self.device is None:
+                center_value = np.mean(np.log(self.search_domain)) if self.log else np.mean(self.search_domain)
+                self.device = Signal(name=self.name, value=center_value)
+
+        # otherwise it must be discrete
+        else:
+            if self.type == "binary":
+                if self.search_domain is None:
+                    self.search_domain = [False, True]
+                if len(self.search_domain) != 2:
+                    raise ValueError("A binary DOF must have a domain of 2.")
+            else:
+                if self.search_domain is None:
+                    raise ValueError("Discrete domain must be supplied for ordinal and categorical degrees of freedom.")
+
+            self.search_domain = set(self.search_domain)
+
+            self.device = Signal(name=self.name, value=list(self.search_domain)[0])
+
         if not self.read_only:
             # check that the device has a put method
             if isinstance(self.device, SignalRO):
                 raise ValueError("You must specify read_only=True for a read-only device.")
 
-        if self.log:
-            if not self.search_bounds[0] > 0:
-                raise ValueError("Search bounds must be strictly positive if log=True.")
-
         # all dof degrees of freedom are hinted
         self.device.kind = "hinted"
 
     @property
-    def _search_bounds(self):
+    def _search_domain(self):
         if self.read_only:
             _readback = self.readback
             return (_readback, _readback)
-        return self.search_bounds
+        return self.search_domain
 
     @property
-    def _trust_bounds(self):
-        if self.trust_bounds is None:
+    def _trust_domain(self):
+        if self.trust_domain is None:
             return (0, np.inf) if self.log else (-np.inf, np.inf)
-        return self.trust_bounds
+        return self.trust_domain
 
     @property
     def readback(self):
+        # there is probably a better way to do this
         return self.device.read()[self.device.name]["value"]
```

Comment:
We typically do it that way. If the `get` method is implemented, it can be simpler to use, but I think we specifically made it to use the `.read()` interface to be able to read a motor value and bundled & individual signals consistently.